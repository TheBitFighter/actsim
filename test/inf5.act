defproc test1(chan?(bool) in; chan!(int<1>) out)
{
  chp {
    *[ [#in]; out!int(in);in? ]
  }
}

defproc boolsrc(chan!(bool) X)
{ 
  chp {
   *[ X!true; X!false ]
  }
}

defproc intsrc(chan!(int<1>) X)
{
  chp {
   *[ X!0; X!1 ]
  }
}

defproc test2(chan?(int<1>) in; chan!(bool) out)
{
  chp {
   *[ [#in];out!bool(in);in? ]
  }
}

defproc boolsink(chan?(bool) W)
{
   bool x;

   chp {
     *[ W?x; log ("got bool: ", x) ]
   }
}

defproc intsink(chan?(int<1>) W)
{
   int<1> x;

   chp {
     *[ W?x; log ("got int: ", x) ]
   }
}

defproc test()
{
   intsrc s;
   test2 t(s.X);
   boolsink ix(t.out);
}
