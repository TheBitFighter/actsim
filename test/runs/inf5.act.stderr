WARNING: boolsink<>: substituting chp model (requested prs, not found)
WARNING: test2<>: substituting chp model (requested prs, not found)
WARNING: intsrc<>: substituting chp model (requested prs, not found)
